source/basic.tex

```diff
@@ -2863,7 +2863,7 @@
 \end{example}
 As a consequence of these rules,
 all declarations of an entity are attached to the same module;
-the entity is said to be \defnx{attached}{attached!entity} to that module.
+the entity is said to be \defnc{attached}{entity} to that module.
```

Comment:
Maybe `\defnctx`; this is really the same thing as `\defncontext` but for an inline definition rather than a Terms and Definitions definition, so using a similar macro seems reasonable.

Reply:
Well, the syntax is a bit different, but you're right about the parallel.  Do you have a suggestion for `\defna`?

Reply:
Well... should `\defna{x}{y}` be an alias for `\defnadjx{x}{}{y}`? (I think we probably always want the *see* entry, so I'm inclined to say yes.) If so, perhaps we could use `\defnadjx{x}{}{y}` directly rather than introducing a new macro?

Reply:
`\defnadjx{x}{}{y}` renders as "_x_ ", which would be problematic at least if punctuation follows.  I'm sure we could teach it to not do so; maybe `#1 #2\unskip` would be enough.

Reply:
Another idea for `\defnctx`: use an optional argument: `\defn[entity]{attached}`. That seems pretty clean. Although now I come to think of it, that might instead be a nice way to get rid of the `\...x` macros instead.

Maybe:
* `\defnx{bodytext}{indextext}` -> `\defn[indextext]{bodytext}`
* `\defnc{bodytext}{indexgroup}` -> `\defnctx[indexgroup]{bodytext}`
* `\defnadjx{bodyadj}{bodynoun}{indexnoun}` -> `\defnadj{bodyadj}[indexnoun]{bodynoun}`
* `\defna{bodyadj}{indexnoun}` -> `\defnadj{bodyadj}[indexnoun]{}`
? (Leaving room for `\defnadj[indexadj]{bodyadj}{bodynoun}` if we ever want that.)

What do you think?

Reply:
In general, that seems like an improvement: fewer magic names and less in the way of argument-order guesswork.  `\newcommand` supports only _leading_ optional arguments, but we could certainly use a raw `\def` for your first `\defnadj`.  We still have to deal with emitting a space only if the `bodynoun` is non-empty.  I don't immediately know how to support optional arguments in two different places, but I'm sure something like `\expandafter` can be used to do it if it ever matters.

Reply:
There's a standard trick for an optional argument after a mandatory argument: `\defnadj` would be a two-parameter command with an optional argument that saves its arguments in some scratch space then expands to another two-parameter command with an optional argument.